### PR TITLE
[Term Entry] Javascript Method: .parseInt()

### LINKED ISSUE #4672: [Term Entry] Javascript Method : .parseInt()
### Reason/inspiration (optional)

We would like a new term entry for JavaScript [Numbers Methods](https://www.codecademy.com/resources/docs/javascript/number-methods) about the `.parseInt()` method. The entry should go in a new file in `docs/content/javascript/concepts/number-methods/terms/parseInt/parseInt.md.`

The entry should:

- Have an Introductory paragraph about what the method is and does.
- Have a syntax section illustrating the function in a   ` ```pseudo code` block.
- An example showing the function in a ` ```js` code block.
- A runnable codebyte example in a ` ```codebyte/javascript` block.

Please refer to the [term entry templates](https://github.com/Codecademy/docs/blob/main/documentation/term-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term  Entry for Existing Topic.

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by andersooi**
### Description

- Added new term entry for Javascript `parseInt()` method

### Issue Solved

Closes #4672 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/javascript/concepts/number-methods/terms/parseInt/parseInt.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+---
+Title: 'parseInt()'
+Description: 'Parses a string argument and returns an integer of the specified radix (base).'
```
Comment (letsbuildanything) [inline content/javascript/concepts/number-methods/terms/parseInt/parseInt.md:3]:
```suggestion
Description: 'Converts a string into an integer.'
```

File: content/javascript/concepts/number-methods/terms/parseInt/parseInt.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+It accepts up to two arguments: the string to be parsed, `myString`, and the radix. The radix is optional, and if unprovided, Javascript will adhere to the following:
+
+- If the `myString` starts with `0x` or `0X`, the radix will be default to 16, and the remaining part of the string will be interpreted as a hexadecimal number.
+- The radix will default to 10 (decimal) for any other input string.
+
+## Syntax
+
+```pseudo code
+parseInt(myString, radix);
```
Comment (letsbuildanything) [inline content/javascript/concepts/number-methods/terms/parseInt/parseInt.md:25]:
```suggestion
parseInt(string, radix);
```

File: content/javascript/concepts/number-methods/terms/parseInt/parseInt.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+```
+
+- `myString` (required): The string to be parsed.
+- `radix` (optional): An integer between 2 and 36 inclusive, representing the base of the numeral system to be used.
```
Comment (letsbuildanything) [inline content/javascript/concepts/number-methods/terms/parseInt/parseInt.md:29]:
```suggestion
- `string`: The string to be parsed. Leading whitespace in the string is ignored.
- `radix` (optional): An integer between 2 and 36 that represents the base of the numeral system to be used. If not provided, the `radix` defaults to 10, except when the string starts with "0x" or "0X" (indicating a hexadecimal number).
```

File: content/javascript/concepts/number-methods/terms/parseInt/parseInt.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+```js
+const x = parseInt('100');
+// x = 100
+
+const y = parseInt('101', 2);
+// y = 5 (binary 101 is 5 in decimal)
+
+const z = parseInt('7F', 16);
+// z = 127 (hexadecimal 7F is 127 in decimal)
```
Comment (letsbuildanything) [inline content/javascript/concepts/number-methods/terms/parseInt/parseInt.md:43]: @andersooi 
print the output of `x`, `y` and `z` using `console.log()` here. Also, Add the output below this code block using ```shell

File: content/javascript/concepts/number-methods/terms/parseInt/parseInt.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+The above example demonstrates parsing a decimal, binary and hexadecimal string into their respective integer values.
+
+The following are a few more examples of unintended use of the `parseInt()` method:
+
+```js
+// invalid radix values
+parseInt('123', 1); // NaN, radix must be between 2 and 36 inclusive
+
+// first character of string
+parseInt('*123'); // NaN, first character of string must be a numeric digit
```
Comment (letsbuildanything) [inline content/javascript/concepts/number-methods/terms/parseInt/parseInt.md:55]: print the output using `console.log()` and add ouput block using ```shell below this. If possible, include this example in the above code block

File: content/javascript/concepts/number-methods/terms/parseInt/parseInt.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+```
+
+> **Note:** The above examples are non-exhaustive.
```
Comment (letsbuildanything) [inline content/javascript/concepts/number-methods/terms/parseInt/parseInt.md:58]: use clear and beginner-friendly terms. Please replace 'non-exhaustive' with a more easy-to-understand term.

File: content/javascript/concepts/number-methods/terms/parseInt/parseInt.md [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+## Codebyte Example
+
+```codebyte/js
+console.log(parseInt('10'));
+console.log(parseInt('101', 2));
+console.log(parseInt('B', 16));
+console.log(parseInt('123abc'));
+console.log(parseInt('abc123'));
```
Comment (letsbuildanything) [inline content/javascript/concepts/number-methods/terms/parseInt/parseInt.md:67]: If possible, please add comments to this code block.

**Review by letsbuildanything — CHANGES_REQUESTED** [review]
@andersooi Thank you for contributing to Codecademy Docs!
I've requested some changes. Please take a moment to address them. If you have any concerns or reasons to reject specific suggestions, feel free to discuss them.

**Comment by andersooi** [comment]
@letsbuildanything I have made the changes based on your comments. Do let me know if there any additional changes that I should make

**Review by letsbuildanything — APPROVED** [review]
@andersooi Thank you for making the changes. It's now ready for the second round of review!

**Review by mamtawardhani — APPROVED** [review]
Hey @andersooi! Thank you for your contribution, the entry looks nice and is ready to get merged 😄

**Pull request merged**